contracts/strategy/TokenizedStrategy.sol

```diff
@@ -184,8 +184,9 @@ contract TokenizedStrategy is ReentrancyGuard {
 
     /// @dev On contract creation we set `asset` for this contract to address(1).
     /// This prevents it from ever being initialized in the future.
-    constructor() {
+    constructor(address _factory) {
         _strategyStorage().asset = ERC20(address(1));
+        FACTORY = _factory;
```

Comment:
This contract is implementation for "proxy" BaseStrategy. It means - this approach isn't supposed to work. If it works now - we have a problem

Reply:
It works due to the nature of immutable variables as discussed on slack